### ISSUE TITLE
Regarding the nitro generator

### ISSUE THREAD

**Issue by Karewen** [therealtear/discord-nitro-generator#9]
Hmm, I have been running the code for some time (24+ hrs) and still no luck. Most of the outputs in my console are either ratelimited proxies or invalid proxies (and some times invalid codes). I made an automatic proxy adder which makes sure my proxies never run out and yes I got it from the proxyscrape API. I am not sure if I am re allygoing to get a working nitro code anytime soon mostly because of the proxies. Anyone got any say about this?

**Comment by HZ-0Z3R0** [comment]
@Karewen The program is somewhat a scam. Look in the closed issues. therealtear deleted my comment saying this: 

You don't really get a valid code every 15 minutes and I can prove it with math. The generator generates a 16 digit alphanumeric string. There's 62 possible combinations for the first letter/digit. Since nitro codes are permutations and not combinations, there are 62^16 possible codes, which is 40000000000000000000000000000 (4 with 28 zeroes). You said the program checks 1 every second. There's 900 seconds in 15 minutes, so you can check 900 codes. Out of the 47 octillion codes, you claim to find a valid one by checking 900. This means there are 4 septillion valid nitro gift codes if you find one every 900 codes! Sorry, this sounds a little implausible.

Basically, you won't get a valid code for a very very very long time, and it's a waste of electricity. The "proof" screenshots is just therealtear buying nitro not generating them.

**Comment by Karewen** [comment]
> @Karewen The program is somewhat a scam. Look in the closed issues. therealtear deleted my comment saying this:
> 
> You don't really get a valid code every 15 minutes and I can prove it with math. The generator generates a 16 digit alphanumeric string. There's 62 possible combinations for the first letter/digit. Since nitro codes are permutations and not combinations, there are 62^16 possible codes, which is 40000000000000000000000000000 (4 with 28 zeroes). You said the program checks 1 every second. There's 900 seconds in 15 minutes, so you can check 900 codes. Out of the 47 octillion codes, you claim to find a valid one by checking 900. This means there are 4 septillion valid nitro gift codes if you find one every 900 codes! Sorry, this sounds a little implausible.
> 
> Basically, you won't get a valid code for a very very very long time, and it's a waste of electricity. The "proof" screenshots is just therealtear buying nitro not generating them.

Your mathematical calculations are correct. However I wouldn't label it as a 'scam' or a 'waste of electricity'. The "get a valid nitro code in 15 minutes" is a bias.

I am using https://repl.it/ to host my program free of cost 24/7. So running the code is not really a problem for me.

Yes, you are correct that I am not going to get a valid code for very very long time considering my chance of getting one is very less (only about 1/62^16). However considering this is free who would not want to try it out? 

With a little improvement to the proxy system the efficiency of the nitro generator can be maximized, which is exactly what I am trying to do now and for that I have contacted tear on discord.

I would suggest you to check repl.it out. It is a very successful site and a much reasonable way to run your script without needing to keep your computer/laptop open for all day.

Have a good day! : )

**Comment by Karewen** [comment]
![image](https://user-images.githubusercontent.com/60471956/103528606-72aabc00-4eae-11eb-9e27-1fadfef2e47d.png)

**Comment by therealtear** [comment]
> @Karewen The program is somewhat a scam. Look in the closed issues. therealtear deleted my comment saying this:
> 
> You don't really get a valid code every 15 minutes and I can prove it with math. The generator generates a 16 digit alphanumeric string. There's 62 possible combinations for the first letter/digit. Since nitro codes are permutations and not combinations, there are 62^16 possible codes, which is 40000000000000000000000000000 (4 with 28 zeroes). You said the program checks 1 every second. There's 900 seconds in 15 minutes, so you can check 900 codes. Out of the 47 octillion codes, you claim to find a valid one by checking 900. This means there are 4 septillion valid nitro gift codes if you find one every 900 codes! Sorry, this sounds a little implausible.
> 
> Basically, you won't get a valid code for a very very very long time, and it's a waste of electricity. The "proof" screenshots is just therealtear buying nitro not generating them.

Quite frankly your a waste of time and electricity

**Comment by therealtear** [comment]
> Hmm, I have been running the code for some time (24+ hrs) and still no luck. Most of the outputs in my console are either ratelimited proxies or invalid proxies (and some times invalid codes). I made an automatic proxy adder which makes sure my proxies never run out and yes I got it from the proxyscrape API. I am not sure if I am re allygoing to get a working nitro code anytime soon mostly because of the proxies. Anyone got any say about this?

If your using bad proxies that error out either change the timeout time to a longer time that matches your proxyscrape api timeout and that is the only reason it’s not working also yes I added the 15 minute thing because I was running a shitty nitro generator scam server that included a token logger I’ll remove it in a future update lol

**Comment by therealtear** [comment]
And I’ll be adding the actual math because I have no reason to fake this at all @HZ-0Z3R0

**Comment by therealtear** [comment]
Also who the hell buys 5 years of nitro which is what I’m at currently

**Comment by Karewen** [comment]
> > Hmm, I have been running the code for some time (24+ hrs) and still no luck. Most of the outputs in my console are either ratelimited proxies or invalid proxies (and some times invalid codes). I made an automatic proxy adder which makes sure my proxies never run out and yes I got it from the proxyscrape API. I am not sure if I am re allygoing to get a working nitro code anytime soon mostly because of the proxies. Anyone got any say about this?
> 
> If your using bad proxies that error out either change the timeout time to a longer time that matches your proxyscrape api timeout and that is the only reason it’s not working also yes I added the 15 minute thing because I was running a shitty nitro generator scam server that included a token logger I’ll remove it in a future update lol

Does the code loop over proxies, currently?

**Comment by therealtear** [comment]
> > > Hmm, I have been running the code for some time (24+ hrs) and still no luck. Most of the outputs in my console are either ratelimited proxies or invalid proxies (and some times invalid codes). I made an automatic proxy adder which makes sure my proxies never run out and yes I got it from the proxyscrape API. I am not sure if I am re allygoing to get a working nitro code anytime soon mostly because of the proxies. Anyone got any say about this?
> > 
> > 
> > If your using bad proxies that error out either change the timeout time to a longer time that matches your proxyscrape api timeout and that is the only reason it’s not working also yes I added the 15 minute thing because I was running a shitty nitro generator scam server that included a token logger I’ll remove it in a future update lol
> 
> Currently, does the code loop over proxies?

No but it’s supposed to
